tests/unit/select_connection_ioloop_tests.py

```diff
@@ -252,39 +293,140 @@ def on_timeout(self):
+        end and reading from the other '''
     def start(self):
+        ''' Create a pair of sockets and poll '''
         self.create_write_socket(self.connected)
         super(IOLoopSimpleMessageTestCaseSelect, self).start()
 
     def connected(self, fd, events, write_only):
+        ''' Respond to 'connected' event by writing to the write-side. '''
         self.assertEqual(events, WRITE)
+        self.assertTrue(write_only is not None)
         logging.debug("Writing to %d message: %s", fd, 'X')
         os.write(fd, b'X')
         self.ioloop.update_handler(fd, 0)
 
     def verify_message(self, msg):
+        ''' Make sure we get what is expected and stop polling '''
         self.assertEqual(msg, b'X')
         self.ioloop.stop()
 
     def start_test(self):
+        ''' Simple message Test'''
         self.start()
 
 class IOLoopSimpleMessageTestCasetPoll(IOLoopSimpleMessageTestCaseSelect):
+    ''' Same as IOLoopSimpleMessageTestCaseSelect but uses 'poll' syscall '''
     SELECT_POLLER = 'poll'
 
 class IOLoopSimpleMessageTestCasetEPoll(IOLoopSimpleMessageTestCaseSelect):
+    ''' Same as IOLoopSimpleMessageTestCaseSelect but uses 'epoll' syscall '''
     SELECT_POLLER = 'epoll'
 
 class IOLoopSimpleMessageTestCasetKqueue(IOLoopSimpleMessageTestCaseSelect):
+    ''' Same as IOLoopSimpleMessageTestCaseSelect but uses 'kqueue' syscall '''
+    SELECT_POLLER = 'kqueue'
+
+
+class DoAfterDelayThread(threading.Thread):
+    '''A class to send signal.SIGUSR1 to cause EINTR in reading thread.'''
+    def __init__(self, initial_wait, func, *args, **kw):
+        self.initial_wait = initial_wait
+        self.func = func
+        super(DoAfterDelayThread, self).__init__(*args, **kw)
+
+    def run(self):
+        time.sleep(self.initial_wait)
+        self.func()
+
+class IOLoopEintrTestCaseSelect(IOLoopBaseTest):
+    ''' Tests if EINTR is properly caught and polling gets resumed. '''
+    SELECT_POLLER = 'select'
+    MSG_CONTENT = b'hello'
+
+    @staticmethod
+    def signal_handler(signum, interrupted_stack):
+        '''A signal handler that gets called in response to
+           os.kill(signal.SIGUSR1).'''
+        pass
+
+    def _eintr_read_handler(self, fileno, events, write_only):
+        '''Read from within poll loop that gets receives eintr error.'''
+        self.assertFalse(write_only)
+        self.assertEqual(events, READ)
+        sock = socket.fromfd(fileno, socket.AF_INET, socket.SOCK_STREAM)
+        mesg = sock.recv(256)
+        self.assertEqual(mesg, self.MSG_CONTENT)
+        self.poller.stop()
+        self._eintr_read_handler_is_called = True
+
+    def _eintr_test_fail(self):
+        '''This function gets called when eintr-test failed to get
+           _eintr_read_handler called.'''
+        self.poller.stop()
+        raise AssertionError('Eintr-test timed out')
+
+    #Disable warning about internal function access.
+    #pylint: disable=W0212
```

Comment:
Don't use the pylint pragma globally, because it disables the checks in other code. Use it on the affected functions or code lines directly instead.